aop/src/main/java/io/micronaut/aop/util/ReactiveMethodInterceptorHelper.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.aop.util;
+
+import io.micronaut.aop.Interceptor;
+import io.micronaut.aop.MethodInvocationContext;
+import io.micronaut.aop.InterceptBuilder;
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.async.publisher.Publishers;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.type.Argument;
+import io.micronaut.core.type.ReturnType;
+import org.reactivestreams.Publisher;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.BiFunction;
+
+/**
+ * Reactive method invocation helper.
+ *
+ * @author Denis stepanov
+ * @since 2.1.0
+ *
+ * @param <T> The declaring type
+ * @param <R> The result of the method call
+ */
+@Internal
+@Experimental
+public class ReactiveMethodInterceptorHelper<T, R> implements InterceptBuilder<T, R> {
```

Comment:
Does this need to public? Probably needs renaming too

Reply:
Renamed it also, do you want to rebase?

Reply:
Makes sense to rebase yes